### PR TITLE
session: add example concurrent usage

### LINKED ISSUE #76: Best practice
I notice there is a `session, err := mgo.Dial(MONGODB_DIAL_URL)` function and also `Clone` and `Copy`.

If we are using MongoDB for a web app backend, I'm wondering what the best practice is out of these options.

1) Call Dial when we need to access database and close as soon as possible (usually with defer within same function).

2) Call Dial at start of request and close at end of request cycle after servicing request.
- Obviously if we must do multiple queries per request, we use the same reference to session _during_
  that request.

3) Call Dial external to request cycle (i.e. when app starts up), store the reference in a global variable and keep using it independent of request and usually never close connection and just let the library do connection pooling.

(Obviously For option 2, an improvement is to dial lazily but that's obvious to readers).
Assuming that we may need to do multiple database queries per request, how exactly do we use `clone` and/or `copy`?

### PR DESCRIPTION
**Pull request by domodwyer**
Fixes #76.

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -86,3 +87,50 @@ func ExampleCredential_x509Authentication() {
 	// You should actually check the error code at each step.
 	_ = err
 }
+
+func ExampleSession_concurrency() {
+	// This example shows the best practise for concurrent use of a mgo session.
+	// 
+	// Internally mgo maintains a connection pool, dialling new connections as
```
Comment (szank) [inline example_test.go]: Is it dialing or dialling ?

File: example_test.go [hunk 2]
```diff
@@ -86,3 +87,50 @@ func ExampleCredential_x509Authentication() {
...
+	}
+
+	///////////////////////////////////////////////
+
+	// Dial a connection to Mongo - this creates the connection pool
+	session, err := Dial("localhost:40003/my_database")
+	if err != nil {
+		panic(err)
+	}
+
```
Comment (szank) [inline example_test.go]: Maybe you could authenticate here to show that all copied sessions inherit the authorization status of the parent session?

File: example_test.go (hunk 1 above)
Comment (domodwyer) [reply example_test.go]: Its dialling, weird eh?

File: example_test.go (hunk 2 above)
Comment (domodwyer) [reply example_test.go]: It's a good idea - but I will leave a comment and update the documentation for Clone. To add the auth in the example the test harness initialisation code would have to be modified.

Comment (domodwyer) [reply example_test.go]: Actually it's already documented behaviour of `Copy()`:

```
Copy works just like New, but preserves the exact authentication information from the original session.
```

**Pull request merged**